### DISCUSSION TITLE
[Feature] URL Parameters

### DISCUSSION THREAD

**Discussion by khaosdoctor** [darrenburns/posting#143] [Ideas]
This is a feature request

# What

Allow to have URL parameters like `some/:id` which would be added in a new section called `parameters` besides `query` and `headers`, this could be automatically picked up by using a regex like `(?:\/)(\:.*)(?:\/?)`.

This would allow us to keep the URLs more configurable

**Comment by Declow** [comment]
I was just looking for the same thing 👍

**Comment by naydichev** [comment]
+1 

This feature would greatly help me to adopt posting as a fulltime replacement for postman :)

**Comment by mattacton** [comment]
+1

This is especially useful for get requests. So far loving the tool, but I'd rather have a single url such as `/users/:id` and then manage the user id in a path parameter / variable section versus continually going up and editing the url.

**Comment by darrenburns** [comment]
I'm wondering if overriding variables/the environment on a per-request basis would solve this for people.

**Comment by bhuynhdev** [comment]
I think path parameters is such a crucial and common part in the HTTP API world that it deserves it own section, like "Query" does. I was checking how Bruno does this: It auto detect when you type a colon followed by slash - like `/users/:user_id/connect` - and then populate a "user_id" item in the Path Parameters section

We can also have a Path Parameters section that is less dynamic: Users needs to define path parameter variables beforehand -> If a curly brace section in the URL - like `/users/{user_id}/connect` - matches one of the path parameters' name then it gets interpolated. Otherwise keep as is

**Comment by darrenburns** [comment]
I'm working on this feature now:

https://github.com/user-attachments/assets/b008db99-33f3-4e95-bcd2-b6b5a7f50f61

**Comment by darrenburns** [comment]
This was implemented and released in 2.9.0. Let me know what you think!
